app/post.go

```diff
@@ -67,6 +86,13 @@ func CreatePost(post *model.Post, teamId string, triggerWebhooks bool) (*model.P
 		pchan = Srv.Store.Post().Get(post.RootId)
 	}
 
+	if utils.IsLicensed() && *utils.Cfg.TeamSettings.ExperimentalTownSquareIsReadOnly &&
+		!post.IsSystemMessage() &&
+		!UserHasPermissionToExecute(post.UserId, model.PERMISSION_MANAGE_SYSTEM) &&
+		Srv.Store.Channel().IsDefaultChannel(post.ChannelId) {
```

Comment:
We should try to avoid having to go to the store to get the channel again here. I don't think that should be a problem though once you rebase since I believe CreatePost now takes the channel itself.